src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -474,6 +474,14 @@ public GHRepository getRepository(String name) throws IOException {
         String[] tokens = name.split("/");
         return retrieve().to("/repos/" + tokens[0] + '/' + tokens[1], GHRepository.class).wrap(this);
     }
+
+    /**
+     * Gets the repository object from its ID
+     */
+    public GHRepository getRepositoryById(String id) throws IOException {
+        return retrieve().to("/repositories/" + id, GHRepository.class).wrap(this);
```

Comment:
@jamesatha Is there this endpoint documented? I was not able to find it [here](https://developer.github.com/v3/repos/).

Reply:
Officially no. But it is supported. The issue filed gives a bit more info about it's "documentation" and how other clients support it: https://github.com/kohsuke/github-api/issues/515